test/test_torch.py

```diff
@@ -1964,6 +1964,13 @@ def test_print(self):
             self.assertExpectedInline(str(x), '''tensor([  100.0000,     0.0100])''')
             torch.set_printoptions(sci_mode=None)  # reset to the default value
 
+            # test show_shape using set_printoptions
```

Comment:
This looks really good but would you make just a few changes to the test:

- try printing a "scalar tensor" with 1 element and 0 dims
- try printing an empty tensor with no elements
- try printing an empty tensor with no elements but multiple dimensions (like a tensor with shape 5, 0, 2)
- try printing a 2D tensor
- Call set_printoptions in a try/finally block and have the finally set `show_shape=False` 

That should ensure the option is unset after this test runs and cover a few more cases

Reply:
Added more tests. :heavy_check_mark:

@mruberry should we add some tests with default settings (```show_shape=True```) too?

Reply:
No this should be OK